Interface features both parts of the ELBO-loss (likelihood part and performance part)

filljonas opened the pull request:
For my recent work about Bayesian Neural Networks for Time Series Forecasting in Hydrology I needed a slight modification of the BLiTZ-library where the function _sample_elbo_ in _variational_estimator.py_ does not only output the total loss, but the likelihood part and performance part individually. In addition, my version outputs the model predictions.

As I think that this could be sensible addition for public use of the library, I would like to apply for a change here.

Best regards,
Jonas Fill

piEsposito commented:
Hi, and thank you for contributing to BLiTZ. I'm gonna run the tests and review the PR.

Tks,
-Pi

piEsposito commented:
The tests did not pass as you changed the return value of a important function of `@variational_estimator`. 

I've presented some approaches you can go to to solve this issue, and you can always create a new function return the values as you wish without changing this function that is a central part of BLiTZ's API. 

Tks!

filljonas commented:
Hi, I restored the original function within @variational_estimator and added my own function where I changed the return type. Please let me know if there is anything else that I need to adjust.

Best,
Jonas

piEsposito commented:
Gonna run the tests and let you know. Thanks again :D

piEsposito commented:
Now it works. I'm merging it, thou it would be nice if you could add the unit tests for this function on other PR.
